lib/protobuf/protoc/generator/message.ex

```diff
@@ -134,10 +143,20 @@ defmodule Protobuf.Protoc.Generator.Message do
       opts: Map.new(opts),
       opts_str: opts_str,
       map: map,
-      oneof: field_desc.oneof_index
+      oneof: field_desc.oneof_index,
+      proto3_optional: field_desc.proto3_optional || false
```

Comment:
Since in the struct this defaults to `false`, when can this field ever be `nil`?

Reply:
In the `Google.Protobuf.FieldDescriptorProto` struct the optional3_optional field is set as an optional field. Because the proto3_optional value obtained from the request defaults to nil, so when the data is mapped, the value will be nil.